Corrects issue with customer refunds and partner lookup for imported lines.

mdietrichc2c opened the pull request:
If you try to auto-complete a customer refund, the partner retrieve will be the partner_id, not the commercial_partner_id like in customer invoices. This fix is to homogenize the behaviour between invoices and refunds.

mdietrichc2c committed:
[FIX] correct partner used in statement import for refunds

pedrobaeza commented on account_statement_transactionid_completion/statement.py:
```diff
@@ -110,7 +110,13 @@ def get_from_transaction_id_and_invoice(self, cr, uid, st_line,
+            if hasattr(invoice, 'commercial_partner_id'):
+                res['partner_id'] = invoice.commercial_partner_id.id
+            else:
+                res['partner_id'] = invoice.partner_id.id
```
I think there is no such 7.1 version, so you can simply put
`res['partner_id'] = invoice.commercial_partner_id.id`

sbidoul replied:
In 7.0 invoice.commercial_partner_id comes from the module account_report_company.

This module is auto_install, but is optional.

mdietrichc2c commented:
Thanks for the feedback.

Sorry, I wanted to write an explanation: this part is derived from the module account_statement_base_completion, which used this comment and this code to use the commercial partner in the invoice, instead of the partner, if it is present. The goal here is to have the same behavior as the "base" one.

I'll re-work the comment, but I'll leave the "fallback" to the partner_id since account_report_company is optional in 7.0.

Regards,
Matthieu Dietrich

pedrobaeza commented:
We can make two things:
- Put _account_report_company_ as a dependency and simplify expression
- Or leave it as you said.

I'm not sure which one I prefer. Maybe the first one, because it assures consistent information.

Regards.

mdietrichc2c commented:
If we have to add it as a dependency, I would prefer to put it in account_statement_base_completion; the same behavior is present, and the module account_statement_transactionid_completion depends on this one.

Although, to be honest, I would prefer not to add a new dependency.

Regards,
Matthieu Dietrich

mdietrichc2c committed:
new comment on last change

pedrobaeza commented:
Yeah, but as you say, in each inherited module you will have to take in account this patch. This dependency is very light, and included on core, so I think it doesn't hurt.

Regards.

mdietrichc2c committed:
[FIX] add dependency on account_report_company + remove unnecessary conditions

mdietrichc2c commented:
Okay, I added the dependency on _account_statement_base_completion_, and removed the condition both in this module and _account_statement_transactionid_completion_.

Regards.

sbidoul commented:
:+1: (code review)

pedrobaeza commented:
Thanks for the changes.

:+1:

gurneyalex commented:
@mdietrichc2c it would be soooooo nice to have an automated test for this fix... :neutral_face:

mdietrichc2c commented:
@gurneyalex I'll do a quick test soon.

mdietrichc2c committed:
[IMP] added unit test for partner fix in completion

mdietrichc2c committed:
[FIX] conflict with most recent version of OCA/7.0

mdietrichc2c committed:
[FIX] still some PEP8 issues

mdietrichc2c committed:
Update .coveragerc to omit test directory

guewen commented:
Hey, coverage decreased :hushed:

Perhaps you should create a test to cover your test :grin:

Seriously, the .coveragerc [0] file miss to omit _/test/_ (I wonder if there is the same issue in the other projects)

[0] https://github.com/mdietrichc2c/bank-statement-reconcile/blob/a71ee406173c9bb84138717d3b28f57c88310e92/.coveragerc

guewen commented:
I'm wrong because it is yaml files. So there is something weird with coveralls.

nbessi commented:
:+1:
